packages/playwright-core/src/utils/processLauncher.ts

```diff
@@ -226,7 +226,7 @@ export async function launchProcess(options: LaunchProcessOptions): Promise<Laun
     killSet.delete(killProcessAndCleanup);
     removeProcessHandlersIfNeeded();
     options.log(`[pid=${spawnedProcess.pid}] <kill>`);
-    if (spawnedProcess.pid && !spawnedProcess.killed && !processClosed) {
```

Comment:
`.killed` means that the process was sent a signal before, not that it was killed. `processClosed` is better for that, especially now that it's set by the `close` event and not the `exit` event. `exit` is fired whenever the process receives a signal.